WooCommerce/Classes/ViewRelated/ReusableViews/UnitInputTableViewCell.swift

```diff
@@ -41,6 +50,25 @@ final class UnitInputTableViewCell: UITableViewCell {
         inputTextField.keyboardType = viewModel.keyboardType
         inputFormatter = viewModel.inputFormatter
         onInputChange = viewModel.onInputChange
+
+        inputAndUnitStackView.removeAllArrangedSubviews()
+        switch viewModel.unitPosition {
+        case .beforeInput, .beforeInputWithoutSpace:
+            inputAndUnitStackView.addArrangedSubviews([unitLabel, inputTextField])
+        case .afterInput, .afterInputWithoutSpace:
+            inputAndUnitStackView.addArrangedSubviews([inputTextField, unitLabel])
+        case .none:
+            inputAndUnitStackView.addArrangedSubviews([inputTextField])
+        }
+
+        switch viewModel.unitPosition {
+        case .beforeInput, .afterInput:
+            inputAndUnitStackView.spacing = Constants.stackViewSpacing
+        case .afterInputWithoutSpace, .beforeInputWithoutSpace:
+            inputAndUnitStackView.spacing = 0
+        default:
+            break
+        }
```

Comment:
Seeing that this is quite long, it's easy to be misinformed in the future that these are unrelated to each other. I believe this deserves its own function to better express the intent:

```suggestion
        rearrangetInputAndUnitStackViewSubviews(using: viewModel.unitPosition)
    }

    private func rearrangetInputAndUnitStackViewSubviews(using position: UnitInputViewModel.UnitPosition) {
        inputAndUnitStackView.removeAllArrangedSubviews()

        switch position {
        case .beforeInput, .beforeInputWithoutSpace:
            inputAndUnitStackView.addArrangedSubviews([unitLabel, inputTextField])
        case .afterInput, .afterInputWithoutSpace:
            inputAndUnitStackView.addArrangedSubviews([inputTextField, unitLabel])
        case .none:
            inputAndUnitStackView.addArrangedSubviews([inputTextField])
        }

        switch position {
        case .beforeInput, .afterInput:
            inputAndUnitStackView.spacing = Constants.stackViewSpacing
        case .afterInputWithoutSpace, .beforeInputWithoutSpace:
            inputAndUnitStackView.spacing = 0
        default:
            break
        }
    }

```